lib.c

```diff
@@ -347,17 +368,35 @@ int osc_init_sdk(struct osc_env *e, const char *profile, unsigned int flag)
 		curl_easy_setopt(e->c, CURLOPT_SSLCERT, cert);
 	if (sslkey)
 		curl_easy_setopt(e->c, CURLOPT_SSLKEY, sslkey);
-	curl_easy_setopt(e->c, CURLOPT_HTTPHEADER, e->headers);
 	curl_easy_setopt(e->c, CURLOPT_WRITEFUNCTION, write_data);
 	curl_easy_setopt(e->c, CURLOPT_USERAGENT, user_agent);
 
 	/* setting CA is CURL_CA_BUNDLE is set */
 	if (ca)
 	  curl_easy_setopt(e->c, CURLOPT_CAINFO, ca);
 
+	e->headers = curl_slist_append(e->headers, "Content-Type: application/json");
+
 	/* For authentification we specify the method and our acces key / secret key */
-	curl_easy_setopt(e->c, CURLOPT_AWS_SIGV4, "osc");
-	curl_easy_setopt(e->c, CURLOPT_USERPWD, ak_sk);
+	if (e->auth_method == OSC_AKSK_METHODE) {
+		curl_easy_setopt(e->c, CURLOPT_AWS_SIGV4, "osc");
+	} else if (e->auth_method == OSC_PASSWORD_METHOD) {
+		time_t clock;
+		struct tm tm;
+		char time_hdr[TIME_HDR_KEY_L + TIMESTAMP_SIZE] = TIME_HDR_KEY;
+
+		time(&clock);
+		if (!gmtime_r(&clock, &tm)) {
+			fprintf(stderr, "gmtime_r fail\n");
+			return -1;
+		}
+		strftime(time_hdr + TIME_HDR_KEY_L - 1,
+			 TIMESTAMP_SIZE, "%Y%m%dT%H%M%SZ", &tm);
+		e->headers = curl_slist_append(e->headers, time_hdr);
+	}
+	curl_easy_setopt(e->c, CURLOPT_HTTPHEADER, e->headers);
+	curl_easy_setopt(e->c, CURLOPT_USERNAME, e->ak);
+	curl_easy_setopt(e->c, CURLOPT_PASSWORD, e->sk);
```

Comment:
Even if `e->auth_method == OSC_AKSK_METHOD` ?

Reply:
yes, ak/sk use CURLOPT_USERNAME/CURLOPT_PASSWORD for the ak and sk.
the `curl_easy_setopt(e->c, CURLOPT_AWS_SIGV4, "osc");` is for v4 specific arguments.